src/client.rs

```diff
@@ -1645,38 +1653,18 @@ where
         pool: &ConnectionPool,
         client_stats: &ClientStats,
     ) -> Result<BytesMut, Error> {
-        if pool.settings.user.statement_timeout > 0 {
-            match tokio::time::timeout(
-                tokio::time::Duration::from_millis(pool.settings.user.statement_timeout),
-                server.recv(),
-            )
-            .await
-            {
-                Ok(result) => match result {
-                    Ok(message) => Ok(message),
-                    Err(err) => {
-                        pool.ban(address, BanReason::MessageReceiveFailed, Some(client_stats));
-                        error_response_terminal(
-                            &mut self.write,
-                            &format!("error receiving data from server: {:?}", err),
-                        )
-                        .await?;
-                        Err(err)
-                    }
-                },
-                Err(_) => {
-                    error!(
-                        "Statement timeout while talking to {:?} with user {}",
-                        address, pool.settings.user.username
-                    );
-                    server.mark_bad();
-                    pool.ban(address, BanReason::StatementTimeout, Some(client_stats));
-                    error_response_terminal(&mut self.write, "pool statement timeout").await?;
-                    Err(Error::StatementTimeout)
-                }
-            }
-        } else {
-            match server.recv().await {
+        let statement_timeout_duration = match pool.settings.user.statement_timeout {
+            0 => tokio::time::Duration::MAX,
+            timeout => tokio::time::Duration::from_millis(timeout),
+        };
+
+        match tokio::time::timeout(
+            statement_timeout_duration,
+            server.recv(Some(&mut self.server_parameters)),
```

Comment:
Are you updating the client-stored parameters based on what the server is sending? I thought this should be the other way around.

Reply:
The parameter status server messages help us to identify what has changed if anything. We're not parsing the queries the client is sending instead we're parsing the result of anything the client sends

Reply:
I guess I'm a bit confused. Doesn't the client send this on startup? Why do we need to catch these `SET` statements during regular querying ops?

Reply:
Check out the varcache in pgbouncer. We were seeing client errors where the client was sending the set `client_encoding` statement and that wasn't being respected because the state was being cleared. This worked back when we were on pgbouncer so we need to this to complete that migration